roles/openshift_node/tasks/main.yml

```diff
@@ -14,7 +14,9 @@
       hostname: "{{ openshift_hostname | default(none) }}"
       public_hostname: "{{ openshift_public_hostname | default(none) }}"
       deployment_type: "{{ openshift_deployment_type }}"
-      dns_ip: "{{ openshift_dns_ip | default(openshift_master_cluster_vip | default(None, true), true) }}"
+      dns_ip: "{{ openshift_dns_ip
+                  | default(openshift_master_cluster_vip
+                  | default(None if openshift.common.version_greater_than_3_1_or_1_1 | bool else first_master_ip | default(None, true), true), true) }}"
```

Comment:
Hmmm, what about an `oo_pick...(var1, var2)` that grabs the first thing defined. The default result could be passed in or `None`.